test/io/test_io.py

```diff
@@ -1648,3 +1650,58 @@ def test_schema_cache_startup_load_with_in_db_config(defaultenv, metapostgrest):
     response = metapostgrest.session.post("/rpc/reset_db_schemas_config")
     assert response.text == ""
     assert response.status_code == 204
+
+
+def test_jwt_cache_size_decreases_after_expiry(defaultenv):
+    "verify that JWT purges expired JWTs"
+
+    relativeSeconds = lambda sec: int(
+        (datetime.now(timezone.utc) + timedelta(seconds=sec)).timestamp()
+    )
+
+    headers = lambda sec: jwtauthheader(
+        {"role": "postgrest_test_author", "exp": relativeSeconds(sec)},
+        SECRET,
+    )
+
+    env = {
+        **defaultenv,
+        "PGRST_JWT_CACHE_MAX_LIFETIME": "86400",
+        "PGRST_JWT_SECRET": SECRET,
+        "PGRST_DB_CONFIG": "false",
+    }
+
+    with run(env=env, port=freeport()) as postgrest:
+
+        # Generate three unique JWT tokens
+        # The 1 second sleep is needed for it generate a unique token
+        hdrs1 = headers(5)
+        postgrest.session.get("/authors_only", headers=hdrs1)
+
+        time.sleep(1)
+
+        hdrs2 = headers(5)
+        postgrest.session.get("/authors_only", headers=hdrs2)
+
+        time.sleep(1)
+
+        hdrs3 = headers(5)
+        postgrest.session.get("/authors_only", headers=hdrs3)
+
+        # the cache should now have three tokens
+        response = postgrest.admin.get("/metrics")
+        assert response.status_code == 200
+        assert "pgrst_jwt_cache_size 3.0" in response.text
```

Comment:
@taimoorzaeem  Is it possible to have the size measured on bytes?

Reply:
Maybe. There is no built-in function for this in `Data.Cache`. I have tried using [https://hackage.haskell.org/package/ghc-datasize](https://hackage.haskell.org/package/ghc-datasize) to calculate size in bytes at runtime, but I have run into problems with that too.

We can try manually writing some logic to calculate the size in bytes, but I guess that would require some effort.

Reply:
Btw, this metric could be a new PR so it's easier to review.